Floofbot/Services/EventLoggerService.cs

```diff
@@ -0,0 +1,503 @@
+﻿using Discord;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Floofbot.Services
+{
+    public class EventLoggerService
+    {
+
+        private FloofDataContext _floofDb;
+        private DiscordSocketClient _client;
+        public EventLoggerService(FloofDataContext floofDb, DiscordSocketClient client)
+        {
+
+            _floofDb = floofDb;
+            _client = client;
+            _client.MessageUpdated += MessageUpdated;
+            _client.MessageDeleted += MessageDeleted;
+            _client.UserBanned += UserBanned;
+            _client.UserUnbanned += UserUnbanned;
+            _client.UserJoined +=UserJoined;
+            _client.UserLeft += UserLeft;
+            _client.GuildMemberUpdated += GuildMemberUpdated;
+        }
+        public async Task<ITextChannel> GetChannel(Discord.IGuild guild, string eventName = null)
+        {
+            if (eventName == null)
+                return null;
+
+            var serverConfig = _floofDb.LogConfigs.Find(guild.Id);
+            System.Reflection.PropertyInfo propertyInfo = serverConfig.GetType().GetProperty(eventName);
+            ulong logChannel = (ulong)(propertyInfo.GetValue(serverConfig, null));
+            var textChannel = await guild.GetTextChannelAsync(logChannel);
+            return textChannel;
+        }
+        public bool IsToggled(IGuild guild)
+        {
+            // check if the logger is toggled on in this server
+            // check the status of logger
+            var ServerConfig = _floofDb.LogConfigs.Find(guild.Id);
+            if (ServerConfig == null) // no entry in DB for server - not configured
+                return false;
+
+            return ServerConfig.IsOn;
+        }
+
+        public async Task MessageUpdated(Cacheable<IMessage, ulong> before, SocketMessage after, ISocketMessageChannel chan)
+        {
+            try
+            {
+                // deal with empty message
+                var messageBefore = (before.HasValue ? before.Value : null) as IUserMessage;
+                if (messageBefore == null)
+                    return;
+
+                var channel = chan as ITextChannel; // channel null, dm message?
+                if (channel == null)
+                    return;
+
+                if (messageBefore.Content == after.Content) // no change
+                    return;
+
+                if ((IsToggled(channel.Guild)) == false) // not toggled on
+                    return;
+
+                Discord.ITextChannel logChannel = await GetChannel(channel.Guild, "MessageEditedChannel");
+                if (logChannel == null)
+                    return;
+
+                var embed = new EmbedBuilder()
+                 .WithTitle($"⚠️ Message Edited | {after.Author.Username}")
```

Comment:
This may be annoying to change on all embeds, but I'd suggest adding the discriminator to the end of the username. If a Discord account is deleted, I believe their ID is lost. It would be easier to search for logs if we could use the typical `<username>#<discriminator>` format instead of just the user ID.